engine/flink/util/src/main/scala/pl/touk/nussknacker/engine/flink/util/metrics/MetricUtils.scala

```diff
@@ -0,0 +1,110 @@
+package pl.touk.nussknacker.engine.flink.util.metrics
+
+import java.util.ServiceLoader
+
+import cats.data.NonEmptyList
+import org.apache.flink.api.common.functions.RuntimeContext
+import org.apache.flink.metrics.{Counter, Gauge, Histogram, Meter, MetricGroup}
+import pl.touk.nussknacker.engine.flink.api.RuntimeContextLifecycle
+
+//Configure this as Service to use new reporting for Flink
+//this is a bit weird way, but it's temporary and passing configuration would be cumbersome
+trait UseNewMetrics
+
+/*
+  IMPORTANT: PLEASE keep Metrics.md up to date
+
+  Handling Flink metrics is a bit tricky. For long time we parsed tags directly in Influx, via graphite plugin
+  This is complex and error prone, so we'd like to switch to passing flink metric variables as tags via native influx API
+  Unfortunately, current Flink Influx report doesn't allow for elastic configuration, so for now
+  we translate by default to `old` way in groupsWithNameForLegacyMode
+ */
+class MetricUtils(runtimeContext: RuntimeContext) {
```

Comment:
I don't like generic *Utils - this name tempting to put here whatever not matches anywere. Maybe better name would be `MetricsCompat`? It is not critical notice, we can merge it with this name also